pkg/manager/manager.go

```diff
@@ -3,13 +3,38 @@ package manager
 import (
 	"context"
 	"fmt"
+	"io"
 
 	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"github.com/kong/go-database-reconciler/pkg/cprint"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
 
+	"github.com/kong/kubernetes-ingress-controller/v3/internal/logging"
 	managerinternal "github.com/kong/kubernetes-ingress-controller/v3/internal/manager"
 	managercfg "github.com/kong/kubernetes-ingress-controller/v3/pkg/manager/config"
 )
 
+// SetupLoggers sets up the loggers for the controller manager.
+func SetupLoggers(c managercfg.Config, output io.Writer) (logr.Logger, error) {
+	zapBase, err := logging.MakeLogger(c.LogLevel, c.LogFormat, output)
+	if err != nil {
+		return logr.Logger{}, fmt.Errorf("failed to make logger: %w", err)
+	}
+	logger := zapr.NewLoggerWithOptions(zapBase, zapr.LogInfoLevel("v"))
+
+	if c.LogLevel != "trace" && c.LogLevel != "debug" {
+		// disable deck's per-change diff output
+		cprint.DisableOutput = true
+	}
+
+	// Prevents controller-runtime from logging
+	// [controller-runtime] log.SetLogger(...) was never called; logs will not be displayed.
+	ctrllog.SetLogger(logger)
+
+	return logger, nil
+}
```

Comment:
I'm unsure if this should be a part of the public API, as it won't be used in KO. It's similar to the `health.NewHealthCheckServer` - we could put it in its own dedicated logging internal package.